pkg/tfgen/generate.go

```diff
@@ -1537,6 +1542,49 @@ func (g *Generator) gatherOverlays() (moduleMap, error) {
 	return modules, nil
 }
 
+func (g *Generator) checkNoDocsError(err error) bool {
+	var e GetRepoPathErr
+	if !errors.As(err, &e) {
+		// Not the right kind of error
+		return false
+	}
+
+	// If we have already warned, we can just discard the message
+	if !g.noDocsRepo {
+		g.logMissingRepoPath(e)
+	}
+	g.noDocsRepo = true
+	return true
+}
+
+func (g *Generator) logMissingRepoPath(err GetRepoPathErr) {
+	msg := `Unable to find the upstream provider's documentation:
+The upstream repository is expected to be at %q.
+%s
+The original error is: %s`
+
+	var correction string
+	if g.info.UpstreamRepoPath != "" {
+		correction = fmt.Sprintf(`
+The upstream repository path has been overridden, but the specified path is invalid.
+You should check the value of:
+tfbridge.ProviderInfo{
+	UpstreamRepoPath: %q,
+}`, g.info.UpstreamRepoPath)
+	} else {
+		correction = fmt.Sprintf(`
+If the expected path is not correct, you should check the values of these fields (current values shown):
+tfbridge.ProviderInfo{
+	GitHubHost:              %q,
+	GitHubOrg:               %q,
+	Name:                    %q,
+	TFProviderModuleVersion: %q,
+}`, g.info.GetGitHubHost(), g.info.GetGitHubOrg(), g.info.Name, g.info.GetProviderModuleVersion())
+	}
+
+	g.sink.Warningf(&diag.Diag{Message: msg}, err.Expected, correction, err.Underlying)
```

Comment:
My only concern is whether this warning will be missed in a pile of other warnings (especially since it's only emitted once). I do think this is a big improvement on the status quo however.

Reply:
There shouldn't be that many warnings. Most warnings are docs warnings and if we see this warning, there will not be other docs warnings (since there are no docs).